staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3562,7 +3562,16 @@ type PodSecurityContext struct {
 	RunAsNonRoot *bool `json:"runAsNonRoot,omitempty" protobuf:"varint,3,opt,name=runAsNonRoot"`
 	// A list of groups applied to the first process run in each container, in addition
 	// to the container's primary GID.  If unspecified, no groups will be added to
-	// any container.
+	// any container. Please remember that the group membership defined in the
+	// container image is kept. That means, if
```

Comment:
is this part of the CRI spec or is it container-runtime implementation dependent?

Reply:
Great question. However, as far as I know, I think this part is also still ambiguous in the CRI spec(v0.25.3): [LinuxContainerSecurityContext.supplemental_groups](https://github.com/kubernetes/cri-api/blob/v0.25.3/pkg/apis/runtime/v1/api.proto#L821-L823), [LinuxSandboxSecurityContext.supplemental_groups](https://github.com/kubernetes/cri-api/blob/v0.25.3/pkg/apis/runtime/v1/api.proto#L317-L319)

```protobuf
message LinuxContainerSecurityContext {
...
    // List of groups applied to the first process run in the container, in
    // addition to the container's primary GID.
    repeated int64 supplemental_groups = 8;
...
}

message LinuxSandboxSecurityContext {
...
    // List of groups applied to the first process run in the sandbox, in
    // addition to the sandbox's primary GID.
    repeated int64 supplemental_groups = 5;
...
}
```

I confirmed contained and crio implemented this behavior. [here](https://github.com/pfnet-research/strict-supplementalgroups-container-runtime/tree/reproduce-bypass-supplementalgroups) is reproduction code.

So, would it be better to update the CRI spec too??

Reply:
> is this part of the CRI spec or is it container-runtime implementation dependent?

I will ask this at sig-node and come back again.

Reply:
@haircommander Thank you for your comment! Then, I will open a PR to add critest in github.com/kubernetes-sigs/cri-tools.

Reply:
I opened a PR in cri-tools repo that adds a test case to critest: https://github.com/kubernetes-sigs/cri-tools/pull/1005

Reply:
I also added an e2e test case in k/k in to test the updated API description: 
https://github.com/kubernetes/kubernetes/pull/113047/commits/9e88694dd8fcae88523be7ff76cf84934066595c